multiplexer/appd/run.go

```diff
@@ -82,50 +76,56 @@ func (a *Appd) Start(args ...string) error {
 	if err := cmd.Start(); err != nil {
 		return fmt.Errorf("failed to start %s: %w", a.path, err)
 	}
+	a.cmd = cmd
+	return nil
+}
 
-	a.pid = cmd.Process.Pid
+func (a *Appd) IsRunning() bool {
+	return !a.IsStopped()
+}
 
-	go func() {
-		// This waits whether process exits naturally or is killed by Stop()
-		if err := cmd.Wait(); err != nil {
-			log.Printf("Process finished with error: %v\n", err)
-		} else {
-			log.Printf("Process finished with no error\n")
-		}
-		a.pid = AppdStopped // Always reset PID when process ends
-	}()
+func (a *Appd) IsStopped() bool {
+	// Never started or failed to start
+	if a.cmd == nil || a.cmd.Process == nil {
```

Comment:
Once we actually stop, does this get nil? Shouldn't we check the Process.Pid being -1 if it isn't nil?

Reply:
> Once we actually stop, does this get nil?

No this never sets `cmd` to nil because the cmd contains `Process` which has information about whether it was stopped (specifically in `cmd.Process.ProcessState`)